lnp2p/src/bolt/bolt1.rs

```diff
@@ -145,45 +145,70 @@ impl Display for Error {
 
 #[cfg(test)]
 mod test {
+    use amplify::hex::FromHex;
     use internet2::TypedEnum;
     use lightning_encoding::LightningDecode;
 
     use super::*;
     use crate::bolt::Messages;
 
     #[test]
-    fn default_init() {
+    fn bolt1_testvec() {
         let init_msg = Messages::Init(Init {
-            global_features: none!(),
-            local_features: default!(),
+            global_features: InitFeatures::default(),
+            local_features: InitFeatures::default(),
             assets: none!(),
             unknown_tlvs: none!(),
         });
-        assert_eq!(&init_msg.serialize(), &[
-            0x00, 0x10, 0x00, 0x01, 0x00, 0x00, 0x01, 0x00
-        ]);
+        assert_eq!(
+            init_msg.serialize(),
+            Vec::<u8>::from_hex("0010000100000100").unwrap()
+        );
+    }
+
+    #[test]
+    fn real_lnp_testvec() {
+        // Real init message sent by lnp-node
+        let init_recv = [0, 16, 0, 0, 0, 0];
+        let msg = Messages::lightning_deserialize(init_recv);
+        // println!("{:?}", msg);
+        assert_eq!(true, msg.is_ok())
     }
 
     #[test]
     fn real_clightning_testvec() {
         // Real init message sent by c-lightning
         let init_recv = [
-            // msg type
-            0u8, 16, //
-            // global features - 2 bytes
-            0, 2, 34, 0, //
-            // local features - 3 bytes
-            0, 3, 2, 170, 162, //
-            // TLV type = 1 (networks / assets)
-            1, //
-            // len
-            32, //
-            // network value
-            111, 226, 140, 10, 182, 241, 179, 114, 193, 166, 162, 70, 174, 99,
-            247, 79, 147, 30, 131, 101, 225, 90, 8, 156, 104, 214, 25, 0, 0, 0,
-            0, 0,
+            0, 16, 0, 2, 33, 0, 0, 7, 8, 160, 0, 8, 10, 105, 162, 1, 32, 6, 34,
+            110, 70, 17, 26, 11, 89, 202, 175, 18, 96, 67, 235, 91, 191, 40,
+            195, 79, 58, 94, 51, 42, 31, 199, 178, 183, 60, 241, 136, 145, 15,
```

Comment:
Again, previous version was specifying which data has which meaning. Can you pls update just the values when needed, leaving the comments?